### DISCUSSION TITLE
ValueError: CTkImage: light_image must be instance if PIL.Image.Image, not <class 'PIL.ImageTk.PhotoImage'>

### DISCUSSION THREAD

**Discussion by edwardjacksonmy** [TomSchimansky/CustomTkinter#2208] [Q&A]
I get an error, by following the instruction in custom tkinter documentation. 

  File "C:\Users\sss\Desktop\Decompile\Delivery Order Entry\Delivery_Order_Entry.py", line 37, in __init__
    self.login.start()
  File "C:\Users\sss\Desktop\Decompile\Delivery Order Entry\Delivery_Order_Entry.py", line 316, in start
    login_image = customtkinter.CTkImage(light_image=login_icon, size=(30, 30))

How am I going to fix? I update the customtkinter library to the latest version still doesn't help a lot.

**Comment by Akascape** [comment]
Don't use PIL.ImageTk.PhotoImage method, use PIL.Image.open

**Reply by edwardjacksonmy** [reply]
This is my original source code. I already change ImageTk.PhotoImage(login_icon) to Image.open(login_icon), it gives me another error. AttributeError: 'Image' object has no attribute 'read'. The conversion from png image to ImageTk, then direct to button image doesn't seem to work, with Image.open and ImageTk.PhotoImage method used.
  
    def start(self):
        login_frame = customtkinter.CTkFrame(master=self.top_frame)
        login_frame.pack(side="right" , pady=8 , padx=(0,30))
        self.user_label = customtkinter.CTkLabel(master=login_frame, text="User Not Logged")
        self.user_label.pack(side="left")
        login_icon = Image.open(r".\icons\login.png").resize((30,30))
        login_icon = ImageTk.PhotoImage(login_icon)
        customtkinter.CTkButton(master=login_frame , text="",fg_color=("gray75", "gray30") , image=login_icon,width=20,
                                command=self.login_window).pack(side="left")
        login_icon = Image.open(r".\icons\logout.png").resize((30,30))
        login_icon = ImageTk.PhotoImage(login_icon)
        customtkinter.CTkButton(master=login_frame , text="",fg_color=("gray75", "gray30") , image=login_icon,width=20,
                                command=self.logout).pack(side="left")

**Reply by Akascape** [reply]
@edwardjacksonmy 
Let me tell you the proper way to import an image:
```python
login_icon = Image.open(r".\icons\login.png") # imported the image with Image.open
login_icon = customtkinter.CTkImage(login_icon, size=(30,30)) # Use CTkImage, not PhotoImage

customtkinter.CTkButton(master=login_frame , text="",fg_color=("gray75", "gray30") , image=login_icon, width=20,
                            command=self.login_window).pack(side="left")
...
```

**Reply by edwardjacksonmy** [reply]
I tried with your method but it seems not working as the same error is out. Any possibility that the library used in the project is not compatible with the version I installed? But the other laptop with the same original source code can work properly.
